@here/harp-mapview/lib/PickHandler.ts

```diff
@@ -149,23 +149,80 @@ export class PickHandler {
         const intersects: THREE.Intersection[] = [];
         const intersectedTiles = this.getIntersectedTiles(rayCaster);
 
+        // This ensures that we check a given dependency only once (because multiple tiles could
+        // have the same dependency).
+        const checkedDependencies = new Set<number>();
+
         for (const { tile, distance } of intersectedTiles) {
             if (pickListener.done && pickListener.furthestResult!.distance < distance) {
                 // Stop when the listener has all results it needs and remaining tiles are further
                 // away than then furthest pick result found so far.
                 break;
             }
-            intersects.length = 0;
-            rayCaster.intersectObjects(tile.objects, true, intersects);
-            for (const intersect of intersects) {
-                pickListener.addResult(this.createResult(intersect));
-            }
+            this.intersectObjects(tile, intersects, pickListener, rayCaster);
+            this.intersectDependentObjects(
+                tile,
+                intersects,
+                pickListener,
+                rayCaster,
+                checkedDependencies
+            );
         }
 
         pickListener.finish();
         return pickListener.results;
     }
 
+    // Adds the list of intersection results to the pick listener. The intersection results are
+    // cleared after being added.
+    private addIntersectResults(intersects: THREE.Intersection[], pickListener: PickListener) {
+        for (const intersect of intersects) {
+            pickListener.addResult(this.createResult(intersect));
+        }
+        intersects.length = 0;
+    }
+
+    // Intersects the tiles objects using the supplied raycaster and adds the intersection results
+    // to the PickListener
+    private intersectObjects(
+        tile: Tile,
+        intersects: THREE.Intersection[],
+        pickListener: PickListener,
+        rayCaster: THREE.Raycaster
+    ) {
+        rayCaster.intersectObjects(tile.objects, true, intersects);
+        this.addIntersectResults(intersects, pickListener);
+    }
```

Comment:
you can get rid of this method, just call rayCaster.intersectObjects directly inside the loop at intersectMapObjects, then call this.intersectDependentObjects, and then call addIntersectResults just once.